bftengine/src/bftengine/SigManager.cpp

```diff
@@ -225,19 +201,23 @@ uint16_t SigManager::getSigLength(PrincipalId pid) const {
   } else {
     std::shared_lock lock(mutex_);
     if (auto pos = verifiers_.find(pid); pos != verifiers_.end()) {
-      return pos->second->signatureLength();
+      auto result = pos->second->signatureLength();
+      return result;
     } else {
       LOG_ERROR(GL, "Unrecognized pid " << pid);
       return 0;
     }
   }
 }
 
-bool SigManager::verifySig(
+bool SigManager::verifyNonReplicaSig(
     PrincipalId pid, const concord::Byte* data, size_t dataLength, const concord::Byte* sig, uint16_t sigLength) const {
+  LOG_DEBUG(GL, "Validating NON-replica with: " << KVLOG(myId_, pid, sigLength));
   bool result = false;
   {
     std::shared_lock lock(mutex_);
+    ConcordAssert(!replicasInfo_.isIdOfReplica(myId_) || !replicasInfo_.isIdOfReplica(pid));
```

Comment:
I think sigManager is used by replicas only, so `replicasInfo_.isIdOfReplica(myId_)` always return true

Reply:
This is not true, it is used by replicas, read only replicas, and previously by the integritychecker.
Someone might try and use it for an incompatible component in the future, this assertion will prevent it.